diff -aurN openbox-3.0/openbox/session.c openbox-2003-12-11/openbox/session.c
--- openbox-3.0/openbox/session.c	2003-10-25 21:39:51.000000000 +0200
+++ openbox-2003-12-11/openbox/session.c	2003-11-11 12:59:50.000000000 +0100
@@ -92,17 +92,10 @@
         prop_res.vals[i].length = strlen(sm_argv[i]);
     }
 
-    if (save_file) {
-        prop_res.vals[i].value = "--sm-save-file";
-        prop_res.vals[i++].length = strlen("--sm-save-file");
-        prop_res.vals[i].value = save_file;
-        prop_res.vals[i++].length = strlen(save_file);
-    } else {
-        prop_res.vals[i].value = "--sm-client-id";
-        prop_res.vals[i++].length = strlen("--sm-client-id");
-        prop_res.vals[i].value = sm_id;
-        prop_res.vals[i++].length = strlen(sm_id);
-    }
+    prop_res.vals[i].value = "--sm-save-file";
+    prop_res.vals[i++].length = strlen("--sm-save-file");
+    prop_res.vals[i].value = save_file;
+    prop_res.vals[i++].length = strlen(save_file);
 
     props[0] = &prop_res;
     props[1] = &prop_cmd;
@@ -169,6 +162,17 @@
 
     if (save_file)
         session_load(save_file);
+    else {
+        gchar *filename;
+
+        /* this algo is from metacity */
+        filename = g_strdup_printf("%d-%d-%u.obs",
+                                   (gint) time(NULL),
+                                   (gint) getpid(),
+                                   g_random_int());
+        save_file = g_build_filename(sm_sessions_path, filename, NULL);
+        g_free(filename);
+    }
 
     sm_argc = *argc;
     sm_argv = *argv;
@@ -219,7 +223,7 @@
         val_hint.value = &hint;
         val_hint.length = 1;
 
-        sprintf(pid, "%ld", (glong)getpid());
+        g_snprintf(pid, sizeof(pid), "%ld", (glong) getpid());
         val_pid.value = pid;
         val_pid.length = strlen(pid);
 
@@ -315,22 +319,10 @@
 
 static gboolean session_save()
 {
-    gchar *filename;
     FILE *f;
     GList *it;
     gboolean success = TRUE;
 
-    if (save_file)
-        unlink(save_file);
-
-    /* this algo is from metacity */
-    filename = g_strdup_printf("%d-%d-%u.obs",
-                               (gint) time(NULL),
-                               (gint) getpid(),
-                               g_random_int());
-    save_file = g_build_filename(sm_sessions_path, filename, NULL);
-    g_free(filename);
-
     f = fopen(save_file, "w");
     if (!f) {
         success = FALSE;


diff -aurN openbox-3.0/render/image.c openbox-2003-12-11/render/image.c
--- openbox-3.0/render/image.c	2003-10-15 05:57:33.000000000 +0200
+++ openbox-2003-12-11/render/image.c	2003-11-11 13:04:47.000000000 +0100
@@ -139,6 +139,9 @@
         dw = (gint)(dh * ((gdouble)sw / sh));
     }
 
+    if (!(dw && dh))
+        return; /* XXX sanity check */
+
     if (sw != dw || sh != dh) {
         /*if (!(rgba->cache && dw == rgba->cwidth && dh == rgba->cheight))*/ {
             g_free(rgba->cache);
